lib/rspotify/user.rb

```diff
@@ -38,6 +38,16 @@ def self.refresh_token(user_id)
       response = RestClient.post(TOKEN_URI, request_body, RSpotify.send(:auth_header))
       response = JSON.parse(response)
       @@users_credentials[user_id]['token'] = response['access_token']
+      proc = @@users_credentials[user_id]['access_refresh_callback']
+      # If the access token expires and a new one is granted via the refresh
+      # token, then this proc will be called with two parameters:
+      # new_access_token and token_lifetime (in seconds)
+      # The purpose is to allow the calling environment to invoke some action,
+      # such as persisting the new access token somewhere, when the new token
+      # is generated.
+      if (!proc.nil?)
```

Comment:
It would be safer to check if it was a proc or responds to `call` as opposed to just not nil
```suggestion
      if (access_refresh_proc.respond_to? :call)
```